src/ddmd/attrib.d

```diff
@@ -499,7 +500,7 @@ extern (C++) final class LinkDeclaration : AttribDeclaration
     {
         super(decl);
         //printf("LinkDeclaration(linkage = %d, decl = %p)\n", p, decl);
-        linkage = p;
+        linkage = (p != LINKsystem) ? p : Target.systemLinkage();
```

Comment:
Yeah, I did, but then I remembered some obscure optimization detail that dmd considers the first branch as hot.  Maybe that's one micro-optimization too far though. ;-)